brats/transformations.py

```diff
@@ -73,3 +79,122 @@ def hist_match(source: np.ndarray, template: np.ndarray) -> np.ndarray:
                                     template_values)
         result[positive] = interp_t_values[bin_idx]
         return result
+
+
+@singledispatch
+def reorder(img: typing.Any):
+    return NotImplemented
```

Comment:
I don't think this is correct usage of `NotImplemented`. It was designed for binary built-in operators. I think you should use `TypeError`.

Reply:
Changed

Reply:
When I think about it now, I don't really like it. We removed raise in favor of typehints to support Duck typing and now we add it again in different place